### PR TITLE
Fixed GaussianNonLinearCalibration

### PR DESCRIPTION
**Pull request by regislebrun**
The estimation of distribution of the observations error was incorrect. It was created using a zero mean to compute the Cholesky factor of its (known) covariance matrix in order to compute the residual function, but was not updated once the residual function was known. It fixes issue #1529.

### CODE REVIEW

File: lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx [hunk 1]
```diff
@@ -330,10 +330,23 @@ void GaussianNonLinearCalibration::run()
 {
   // Compute the posterior MAP
   // Error distribution
-  const Normal error(Point(errorCovariance_.getDimension()), errorCovariance_);
+  // It is built in two steps to benefit from the Cholsky factorization of the
```
Comment (josephmure) [inline lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx:333]:
```suggestion
  // It is built in two steps to benefit from the Cholesky factorization of the
```

**Comment by josephmure** [comment]
The gap between a value produced with VisualStudio and the corresponding expected value can be quite large:
[-0.0842 instead of -0.0855](https://ci.appveyor.com/project/openturns/openturns/builds/33787714/job/x5c13dhxx413kd4c#L6895) for example. Is the optimization algorithm unstable?

**Comment by jschueller** [comment]
see #1588

**Comment by mbaudin47** [comment]
To sum up the change the updated GaussianNonLinearCalibration class has a resulting error distribution which is Normal with a nonzero mean and with a covariance matrix equal to the errorCovariance input argument. This corresponds to the fact that the linearization of the function near the optimum makes so that the residuals may not necessarily have a zero mean: this effect can indeed be retrieved with the residuals.getMean() method. 

In order to check the code, we can use the GaussianNonLinearCalibration with a flat prior (i.e. a Normal with a large sigma). We must get the same results as the NonLinearLeastSquaresCalibration.

**Pull request closed without merging**